powerauth-push-server/src/main/java/io/getlime/push/service/PushMessageSenderService.java

```diff
@@ -194,14 +197,20 @@ public void sendCampaignMessage(String appId, Platform platform, String token, P
      * @throws PushServerException In case any issue happens while sending the push message. Detailed information about
      * the error can be found in exception message.
      */
-    public void sendCampaignMessage(final String appId, Platform platform, final String token, PushMessageBody pushMessageBody, PushMessageAttributes attributes, Priority priority, String userId, Long deviceId, String activationId) throws PushServerException {
+    public void sendCampaignMessage(final String appId, final Platform platform, final ApnsEnvironment environment, final String token, final PushMessageBody pushMessageBody, final PushMessageAttributes attributes, final Priority priority, final String userId, final Long deviceId, final String activationId) throws PushServerException {
         final AppRelatedPushClient pushClient = prepareClients(appId);
 
         final PushMessageEntity pushMessageObject = pushMessageDAO.storePushMessageObject(pushMessageBody, attributes, userId, activationId, deviceId);
 
         switch (platform) {
-            case IOS, APNS ->
-                    pushSendingWorker.sendMessageToApns(pushClient.getApnsClient(), pushMessageBody, attributes, priority, token, pushClient.getAppCredentials().getApnsBundle(), createPushSendingCallback(token, pushMessageObject, pushClient));
+            case IOS, APNS -> {
+                final ApnsEnvironment apnsEnvironment = environment != null ? resolveApnsEnvironment(environment.getKey()) : resolveApnsEnvironment(null);
+                if (apnsEnvironment == null) {
+                    logger.error("Campaign push message cannot be sent because APNs environment configuration failed. Check configuration of application property 'powerauth.push.service.apns.useDevelopment'.");
```

Comment:
🤔 Shouldn't we return from this branch or wrap the next lines into an `else` branch?

Reply:
Yes, good catch.